server/fleet/software_installer.go

```diff
@@ -35,6 +35,22 @@ func (FailingSoftwareInstallerStore) Exists(ctx context.Context, installerID str
 	return false, errors.New("software installer store not properly configured")
 }
 
+// SoftwareInstallDetailsResult contains all of the information
+// required for a client to pull in and install software from the fleet server
+type SoftwareInstallDetails struct {
+	HostID uint `json:"-" db:"host_id"`
+	// ExecutionID is a unique identifier for this installation
+	ExecutionID string `json:"execution_id" db:"execution_id"`
+	// InstallerID is the unique identifier for the software package metadata in Fleet.
+	InstallerID uint `json:"installer_id" db:"installer_id"`
+	// PreInstallCondition is the query to run as a condition to installing the software package.
+	PreInstallCondition string `json:"pre_install_condition" db:"pre_install_condition"`
+	// InstallScript is the script to run to install the software package.
+	InstallScript []byte `json:"install_script" db:"install_script"`
+	// PostInstallScript is the script to run after installing the software package.
+	PostInstallScript []byte `json:"post_install_script" db:"post_install_script"`
```

Comment:
I think we want those as strings, not `[]byte`? It will be non-binary strings, and having them as `[]byte` will bloat the payload by base64-encoding them.

Reply:
The base64 payload was part of the spec but I'd be happy to replace it with a string

Reply:
Yeah I think we should, just as we did here: https://github.com/fleetdm/fleet/pull/18689#discussion_r1586747068. @roperzh to double-check there was not a specific reason to have it base64-encoded here?

Reply:
This is my fault, I forgot to update the issue with the spec after pinging you Dante in that comment. It's updated now to be a string.

Sorry about that!